src/aap_eda/core/models/setting.py

```diff
@@ -0,0 +1,30 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from django.db import models
+
+from aap_eda.core.utils.crypto.fields import EncryptedTextField
+
+__all__ = ("Setting",)
+
+
+class Setting(models.Model):
+    class Meta:
+        db_table = "core_setting"
+        indexes = [models.Index(fields=["key"])]
+
+    key = models.CharField(max_length=255)
```

Comment:
Is there a reason to not enforce the uniqueness of the key at the DB layer? I don't see a reason why we should allow duplicates here.

Reply:
Later when this table is expanded to store settings per user we will see key duplication. I added the uniqueness for now. Can lift it up when the time comes.